### PR TITLE
feat(rpc): implement Filecoin.StateMinerPreCommitDepositForPower

### LINKED ISSUE #4214: StateMinerPreCommitDepositForPower

### PR DESCRIPTION
**Pull request by sudo-shashank**
## Summary of changes

Changes introduced in this pull request:

- Implemented `Filecoin.StateMinerPreCommitDepositForPower` RPC method
```
| RPC Method                                      | Forest | Lotus |
|-------------------------------------------------|--------|-------|
| Filecoin.StateMinerPreCommitDepositForPower (5) | Valid  | Valid |
```

## Reference issue to close (if applicable)

Closes #4214 

## Change checklist

- [X] I have performed a self-review of my own code,
- [X] I have made corresponding changes to the documentation,
- [X] I have added tests that prove my fix is effective or that my feature works (if possible),
- [X] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by elmattic — APPROVED** [review]

**Comment by elmattic** [comment]
Good!

File: src/rpc/methods/state.rs [hunk 1]
```diff
@@ -776,6 +777,70 @@ impl RpcMethod<3> for StateMinerInitialPledgeCollateral {
...
+        let sector_weight =
+            if ctx.state_manager.get_network_version(ts.epoch()) < NetworkVersion::V16 {
+                qa_power_for_weight(sector_size, duration, &w, &vw)
+            } else {
+                qa_power_max(sector_size)
+            };
+
+        let actor = ctx
+            .state_manager
+            .get_actor(&Address::POWER_ACTOR, state)?
```
Comment (hanabi1224) [inline src/rpc/methods/state.rs:826]: Let's use `.get_required_actor` to save the `.context` call

File: src/rpc/methods/state.rs [hunk 2]
```diff
@@ -776,6 +777,70 @@ impl RpcMethod<3> for StateMinerInitialPledgeCollateral {
...
+            .context("Power actor address could not be resolved")?;
+        let power_state = power::State::load(ctx.store(), actor.code, actor.state)?;
+        let power_smoothed = power_state.total_power_smoothed();
+
+        let actor = ctx
+            .state_manager
+            .get_actor(&Address::REWARD_ACTOR, state)?
```
Comment (hanabi1224) [inline src/rpc/methods/state.rs:833]: Ditto `.get_required_actor`

File: src/rpc/methods/state.rs [hunk 3]
```diff
@@ -776,6 +777,70 @@ impl RpcMethod<3> for StateMinerInitialPledgeCollateral {
...
+        let state = *ts.parent_state();
+
+        let sector_size = pci
+            .seal_proof
+            .sector_size()
+            .map_err(|e| anyhow::anyhow!("failed to get resolve size: {e}"))?;
+
+        let actor = ctx
+            .state_manager
+            .get_actor(&Address::MARKET_ACTOR, state)?
```
Comment (hanabi1224) [inline src/rpc/methods/state.rs:806]: Ditto `.get_required_actor`

File: src/rpc/methods/state.rs (hunk 2 above)
Comment (sudo-shashank) [reply src/rpc/methods/state.rs:833]: fixed

File: src/rpc/methods/state.rs (hunk 3 above)
Comment (sudo-shashank) [reply src/rpc/methods/state.rs:806]: fixed

**Review by hanabi1224 — APPROVED** [review]

**Pull request merged**